storage/tree_validation_test.go

```diff
@@ -162,8 +163,11 @@ func TestValidateTreeForCreation(t *testing.T) {
 	}
 	for i, test := range tests {
 		err := ValidateTreeForCreation(test.tree)
-		if hasErr := err != nil; hasErr != test.wantErr {
+		switch hasErr := err != nil; {
+		case hasErr != test.wantErr:
 			t.Errorf("%v: ValidateTreeForCreation() = %v, wantErr = %v", i, err, test.wantErr)
+		case hasErr && errors.ErrorCode(err) != errors.InvalidArgument:
+			t.Errorf("%v: ValidateTreeForCreation() = %v, wantCode = %v", i, err, errors.InvalidArgument)
 		}
```

Comment:
Personal preference, but I'd find this easier to read as if/elseif:

```go
if hasErr := err != nil; hasErr != test.wantErr {
} else if hasErr && errors.ErrorCode(err) != errors.InvalidArgument {
}
```
(below also)